frontend/src/features/permits/components/permit-list/ApplicationsInReviewRowOptions.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useState } from "react";
+import { OnRouteBCTableRowActions } from "../../../../common/components/table/OnRouteBCTableRowActions";
+import { useUpdateApplicationQueueStatusMutation } from "../../hooks/hooks";
+import { CASE_ACTIVITY_TYPES } from "../../types/CaseActivityType";
+import { ApplicationInReviewModal } from "./ApplicationInReviewModal";
+import { useQueryClient } from "@tanstack/react-query";
+
+const PERMIT_ACTION_OPTION_TYPES = {
+  WITHDRAW_APPLICATION: "withdrawApplication",
+} as const;
+
+type PermitActionOptionType =
+  (typeof PERMIT_ACTION_OPTION_TYPES)[keyof typeof PERMIT_ACTION_OPTION_TYPES];
+
+const getOptionLabel = (optionType: PermitActionOptionType): string => {
+  if (optionType === PERMIT_ACTION_OPTION_TYPES.WITHDRAW_APPLICATION) {
+    return "Withdraw Application";
+  }
+
+  return "";
+};
+
+const ALL_OPTIONS = [
+  {
+    label: getOptionLabel(PERMIT_ACTION_OPTION_TYPES.WITHDRAW_APPLICATION),
+    value: PERMIT_ACTION_OPTION_TYPES.WITHDRAW_APPLICATION,
+  },
+];
+
+const getOptions = (isInReview: boolean) => {
+  return ALL_OPTIONS.filter((option) => {
+    // Exclude 'WITHDRAW_APPLICATION' if 'isInReview' is false
+    if (
+      isInReview &&
+      option.value === PERMIT_ACTION_OPTION_TYPES.WITHDRAW_APPLICATION
+    ) {
+      return false;
+    }
+    return true;
+  });
+};
+
+export const ApplicationsInReviewRowOptions = ({
+  isInReview,
+  permitId,
+}: {
+  isInReview: boolean;
+  permitId: string;
+}) => {
+  const [isAIRModalOpen, setIsAIRModalOpen] = useState<boolean>(false);
+
+  const queryClient = useQueryClient();
+
+  const handleCloseAIRModal = () => {
+    setIsAIRModalOpen(false);
+    queryClient.invalidateQueries({
+      queryKey: ["applicationsInQueue"],
+    });
+  };
+
+  const { mutateAsync, isError, error } =
+    useUpdateApplicationQueueStatusMutation();
+
+  useEffect(() => {
+    if (isError && error.response?.status === 422) {
```

Comment:
How is this supposed to work here? We only show the modal when the error status is 422? How should the other non-422 errors be handled?